int/api/cmd/deploySC.go

```diff
@@ -45,22 +52,18 @@ func (d *deploySC) Handle(params operations.CmdDeploySCParams) middleware.Respon
 				})
 	}
 
-	if len(decodedDatastore) == 0 {
-		decodedDatastore = nil
-	}
-
 	operationResponse, events, err := onchain.DeploySC(
 		d.networkInfos,
-		params.WalletNickname,
-		*params.GasLimit,
-		*params.Coins,
-		*params.Fee,
-		*params.Expiry,
-		file,
-		decodedDatastore,
-		sendoperation.OperationBatch{NewBatch: false, CorrelationID: ""},
+		params.Body.Nickname,
+		sendoperation.MaxGasAllowedExecuteSC, // default
+		*params.Body.MaxCoins, // maxCoins 
+		*params.Body.Coins,    // smart contract deployment "fee"
+		sendoperation.DefaultExpiryInSlot,
+		parameters, 
+		smartContractByteCode,
+		deployerSCByteCode,
 		&signer.WalletPlugin{},
-		"",
+		"Deploying website",
```

Comment:
Deploying contract ?? 
Maybe you can add a "description" field to the endpoint to make it dynamic